Samples/PDESampleApplication/Form2.cs

```diff
@@ -0,0 +1,319 @@
+            }
+        }
+
+        async void ProtectFolderRecursively(StorageFolder folder, UserDataAvailability level)
+        {
+            // Protect the folder first so new files / folders after this point will
+            // get protected automatically.
+            ProtectAndLog(folder, level);
+
+            // Protect all sub-folders recursively.
+            var subFolders = await folder.GetFoldersAsync();
+            foreach (var subFolder in subFolders)
+            {
+                ProtectFolderRecursively(subFolder, level);
+            }
+
+            // Finally protect all existing files in the folder.
+            var files = await folder.GetFilesAsync();
+            foreach (var file in files)
+            {
+                ProtectAndLog(file, level);
+            }
+        }
+
+        async void UnprotectBuffer(String g_protectbase64EncodedContent)
+        {
+            try
+            {
+                var protectedBuffer = CryptographicBuffer.DecodeFromBase64String(g_protectbase64EncodedContent);
+                var result = await g_udpm.UnprotectBufferAsync(protectedBuffer);
+                if (result.Status == UserDataBufferUnprotectStatus.Succeeded)
+                {
+                    LogLine("Result of Unprotecting the buffer:" +
+                        CryptographicBuffer.ConvertBinaryToString(BinaryStringEncoding.Utf8, result.UnprotectedBuffer));
+                    LogLine("Status of Unprotectng the buffer:" + result.Status);
+                }
+                else
+                {
+                    LogLine("This protected buffer is currently unavailable for unprotection");
+                }
+            }
+            catch(ArgumentException)
+            {
+                LogLine("Invalid Argument, please check text being unprotected!");
+            }
+            catch (Exception)
+            {
+                LogLine("Something went wrong");
+            }
+        }
+
+        async void ProtectBuffer(String text, UserDataAvailability level)
+        {
+            try
+            {
+                var buffer = CryptographicBuffer.ConvertStringToBinary(text, BinaryStringEncoding.Utf8);
+                g_protectedContent = await g_udpm.ProtectBufferAsync(buffer, level);
+                g_protectbase64EncodedContent = CryptographicBuffer.EncodeToBase64String(g_protectedContent);
+                LogLine("Protected buffer: " + g_protectbase64EncodedContent);
+            }
+            catch (ArgumentException)
+            {
+                LogLine("Invalid Argument, please check text being protected!");
+            }
+            catch (Exception)
+            {
+                LogLine("Something went wrong");
+            }
+        }
+        private void button4_Click(object sender, EventArgs e)
+        {
+            if(folderBrowserDialog1.ShowDialog() == DialogResult.OK)
+            {
+                listView1.Items.Add(folderBrowserDialog1.SelectedPath.Trim());
```

Comment:
Still a lot of UI element names that are too generic (like listView1, folderBrowserDialog1). Those should have names appropriate to what they're being used for in the UI.